components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java

```diff
@@ -1051,6 +1055,65 @@ public void updateWsdl(String resourcePath, String wsdlDefinition) throws APIMan
         }
     }
 
+    /**
+     * Returns the graphQL content in registry specified by the wsdl name
+     *
+     * @param apiId Api Identifier
+     * @return graphQL content matching name if exist else null
+     */
+    @Override
+    public String getGraphqlSchema(APIIdentifier apiId) throws APIManagementException {
+        String schemaDoc = null;
+        String schemaName = apiId.getProviderName() + APIConstants.GRAPHQL_SCHEMA_PROVIDER_SEPERATOR + apiId.getApiName() +
+                apiId.getVersion() + APIConstants.GRAPHQL_SCHEMA_FILE_EXTENSION;
+        String schemaResourePath = APIConstants.API_GRAPHQL_SCHEMA_RESOURCE_LOCATION + schemaName;
+        try {
+            if (registry.resourceExists(schemaResourePath)) {
+                Resource schemaResource = registry.get(schemaResourePath);
+                schemaDoc = IOUtils.toString(schemaResource.getContentStream(),
+                        RegistryConstants.DEFAULT_CHARSET_ENCODING);
+            }
+        } catch (RegistryException e) {
+            String msg = "Error while getting schema file from the registry ";
```

Comment:
Shall we add the context information for the log. Ex: schemaResourePath

Reply:
Done